Index: PadicoTM/autogen.sh
===================================================================
--- PadicoTM/autogen.sh	(révision 5179)
+++ PadicoTM/autogen.sh	(copie de travail)
@@ -1,6 +1,6 @@
 #! /bin/sh -e
 
-export M4PATH=../building-tools:./building-tools:${M4PATH}
+export M4PATH=./building-tools:./building-tools:${M4PATH}
 
 ${AUTOCONF:-autoconf} 
 ${AUTOHEADER:-autoheader} -f
Index: PadicoTM/building-tools/COPYING
===================================================================
--- PadicoTM/building-tools/COPYING	(nonexistent)
+++ PadicoTM/building-tools/COPYING	(copie de travail)
@@ -0,0 +1,339 @@
+		    GNU GENERAL PUBLIC LICENSE
+		       Version 2, June 1991
+
+ Copyright (C) 1989, 1991 Free Software Foundation, Inc.
+                          675 Mass Ave, Cambridge, MA 02139, USA
+ Everyone is permitted to copy and distribute verbatim copies
+ of this license document, but changing it is not allowed.
+
+			    Preamble
+
+  The licenses for most software are designed to take away your
+freedom to share and change it.  By contrast, the GNU General Public
+License is intended to guarantee your freedom to share and change free
+software--to make sure the software is free for all its users.  This
+General Public License applies to most of the Free Software
+Foundation's software and to any other program whose authors commit to
+using it.  (Some other Free Software Foundation software is covered by
+the GNU Library General Public License instead.)  You can apply it to
+your programs, too.
+
+  When we speak of free software, we are referring to freedom, not
+price.  Our General Public Licenses are designed to make sure that you
+have the freedom to distribute copies of free software (and charge for
+this service if you wish), that you receive source code or can get it
+if you want it, that you can change the software or use pieces of it
+in new free programs; and that you know you can do these things.
+
+  To protect your rights, we need to make restrictions that forbid
+anyone to deny you these rights or to ask you to surrender the rights.
+These restrictions translate to certain responsibilities for you if you
+distribute copies of the software, or if you modify it.
+
+  For example, if you distribute copies of such a program, whether
+gratis or for a fee, you must give the recipients all the rights that
+you have.  You must make sure that they, too, receive or can get the
+source code.  And you must show them these terms so they know their
+rights.
+
+  We protect your rights with two steps: (1) copyright the software, and
+(2) offer you this license which gives you legal permission to copy,
+distribute and/or modify the software.
+
+  Also, for each author's protection and ours, we want to make certain
+that everyone understands that there is no warranty for this free
+software.  If the software is modified by someone else and passed on, we
+want its recipients to know that what they have is not the original, so
+that any problems introduced by others will not reflect on the original
+authors' reputations.
+
+  Finally, any free program is threatened constantly by software
+patents.  We wish to avoid the danger that redistributors of a free
+program will individually obtain patent licenses, in effect making the
+program proprietary.  To prevent this, we have made it clear that any
+patent must be licensed for everyone's free use or not licensed at all.
+
+  The precise terms and conditions for copying, distribution and
+modification follow.
+
+		    GNU GENERAL PUBLIC LICENSE
+   TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
+
+  0. This License applies to any program or other work which contains
+a notice placed by the copyright holder saying it may be distributed
+under the terms of this General Public License.  The "Program", below,
+refers to any such program or work, and a "work based on the Program"
+means either the Program or any derivative work under copyright law:
+that is to say, a work containing the Program or a portion of it,
+either verbatim or with modifications and/or translated into another
+language.  (Hereinafter, translation is included without limitation in
+the term "modification".)  Each licensee is addressed as "you".
+
+Activities other than copying, distribution and modification are not
+covered by this License; they are outside its scope.  The act of
+running the Program is not restricted, and the output from the Program
+is covered only if its contents constitute a work based on the
+Program (independent of having been made by running the Program).
+Whether that is true depends on what the Program does.
+
+  1. You may copy and distribute verbatim copies of the Program's
+source code as you receive it, in any medium, provided that you
+conspicuously and appropriately publish on each copy an appropriate
+copyright notice and disclaimer of warranty; keep intact all the
+notices that refer to this License and to the absence of any warranty;
+and give any other recipients of the Program a copy of this License
+along with the Program.
+
+You may charge a fee for the physical act of transferring a copy, and
+you may at your option offer warranty protection in exchange for a fee.
+
+  2. You may modify your copy or copies of the Program or any portion
+of it, thus forming a work based on the Program, and copy and
+distribute such modifications or work under the terms of Section 1
+above, provided that you also meet all of these conditions:
+
+    a) You must cause the modified files to carry prominent notices
+    stating that you changed the files and the date of any change.
+
+    b) You must cause any work that you distribute or publish, that in
+    whole or in part contains or is derived from the Program or any
+    part thereof, to be licensed as a whole at no charge to all third
+    parties under the terms of this License.
+
+    c) If the modified program normally reads commands interactively
+    when run, you must cause it, when started running for such
+    interactive use in the most ordinary way, to print or display an
+    announcement including an appropriate copyright notice and a
+    notice that there is no warranty (or else, saying that you provide
+    a warranty) and that users may redistribute the program under
+    these conditions, and telling the user how to view a copy of this
+    License.  (Exception: if the Program itself is interactive but
+    does not normally print such an announcement, your work based on
+    the Program is not required to print an announcement.)
+
+These requirements apply to the modified work as a whole.  If
+identifiable sections of that work are not derived from the Program,
+and can be reasonably considered independent and separate works in
+themselves, then this License, and its terms, do not apply to those
+sections when you distribute them as separate works.  But when you
+distribute the same sections as part of a whole which is a work based
+on the Program, the distribution of the whole must be on the terms of
+this License, whose permissions for other licensees extend to the
+entire whole, and thus to each and every part regardless of who wrote it.
+
+Thus, it is not the intent of this section to claim rights or contest
+your rights to work written entirely by you; rather, the intent is to
+exercise the right to control the distribution of derivative or
+collective works based on the Program.
+
+In addition, mere aggregation of another work not based on the Program
+with the Program (or with a work based on the Program) on a volume of
+a storage or distribution medium does not bring the other work under
+the scope of this License.
+
+  3. You may copy and distribute the Program (or a work based on it,
+under Section 2) in object code or executable form under the terms of
+Sections 1 and 2 above provided that you also do one of the following:
+
+    a) Accompany it with the complete corresponding machine-readable
+    source code, which must be distributed under the terms of Sections
+    1 and 2 above on a medium customarily used for software interchange; or,
+
+    b) Accompany it with a written offer, valid for at least three
+    years, to give any third party, for a charge no more than your
+    cost of physically performing source distribution, a complete
+    machine-readable copy of the corresponding source code, to be
+    distributed under the terms of Sections 1 and 2 above on a medium
+    customarily used for software interchange; or,
+
+    c) Accompany it with the information you received as to the offer
+    to distribute corresponding source code.  (This alternative is
+    allowed only for noncommercial distribution and only if you
+    received the program in object code or executable form with such
+    an offer, in accord with Subsection b above.)
+
+The source code for a work means the preferred form of the work for
+making modifications to it.  For an executable work, complete source
+code means all the source code for all modules it contains, plus any
+associated interface definition files, plus the scripts used to
+control compilation and installation of the executable.  However, as a
+special exception, the source code distributed need not include
+anything that is normally distributed (in either source or binary
+form) with the major components (compiler, kernel, and so on) of the
+operating system on which the executable runs, unless that component
+itself accompanies the executable.
+
+If distribution of executable or object code is made by offering
+access to copy from a designated place, then offering equivalent
+access to copy the source code from the same place counts as
+distribution of the source code, even though third parties are not
+compelled to copy the source along with the object code.
+
+  4. You may not copy, modify, sublicense, or distribute the Program
+except as expressly provided under this License.  Any attempt
+otherwise to copy, modify, sublicense or distribute the Program is
+void, and will automatically terminate your rights under this License.
+However, parties who have received copies, or rights, from you under
+this License will not have their licenses terminated so long as such
+parties remain in full compliance.
+
+  5. You are not required to accept this License, since you have not
+signed it.  However, nothing else grants you permission to modify or
+distribute the Program or its derivative works.  These actions are
+prohibited by law if you do not accept this License.  Therefore, by
+modifying or distributing the Program (or any work based on the
+Program), you indicate your acceptance of this License to do so, and
+all its terms and conditions for copying, distributing or modifying
+the Program or works based on it.
+
+  6. Each time you redistribute the Program (or any work based on the
+Program), the recipient automatically receives a license from the
+original licensor to copy, distribute or modify the Program subject to
+these terms and conditions.  You may not impose any further
+restrictions on the recipients' exercise of the rights granted herein.
+You are not responsible for enforcing compliance by third parties to
+this License.
+
+  7. If, as a consequence of a court judgment or allegation of patent
+infringement or for any other reason (not limited to patent issues),
+conditions are imposed on you (whether by court order, agreement or
+otherwise) that contradict the conditions of this License, they do not
+excuse you from the conditions of this License.  If you cannot
+distribute so as to satisfy simultaneously your obligations under this
+License and any other pertinent obligations, then as a consequence you
+may not distribute the Program at all.  For example, if a patent
+license would not permit royalty-free redistribution of the Program by
+all those who receive copies directly or indirectly through you, then
+the only way you could satisfy both it and this License would be to
+refrain entirely from distribution of the Program.
+
+If any portion of this section is held invalid or unenforceable under
+any particular circumstance, the balance of the section is intended to
+apply and the section as a whole is intended to apply in other
+circumstances.
+
+It is not the purpose of this section to induce you to infringe any
+patents or other property right claims or to contest validity of any
+such claims; this section has the sole purpose of protecting the
+integrity of the free software distribution system, which is
+implemented by public license practices.  Many people have made
+generous contributions to the wide range of software distributed
+through that system in reliance on consistent application of that
+system; it is up to the author/donor to decide if he or she is willing
+to distribute software through any other system and a licensee cannot
+impose that choice.
+
+This section is intended to make thoroughly clear what is believed to
+be a consequence of the rest of this License.
+
+  8. If the distribution and/or use of the Program is restricted in
+certain countries either by patents or by copyrighted interfaces, the
+original copyright holder who places the Program under this License
+may add an explicit geographical distribution limitation excluding
+those countries, so that distribution is permitted only in or among
+countries not thus excluded.  In such case, this License incorporates
+the limitation as if written in the body of this License.
+
+  9. The Free Software Foundation may publish revised and/or new versions
+of the General Public License from time to time.  Such new versions will
+be similar in spirit to the present version, but may differ in detail to
+address new problems or concerns.
+
+Each version is given a distinguishing version number.  If the Program
+specifies a version number of this License which applies to it and "any
+later version", you have the option of following the terms and conditions
+either of that version or of any later version published by the Free
+Software Foundation.  If the Program does not specify a version number of
+this License, you may choose any version ever published by the Free Software
+Foundation.
+
+  10. If you wish to incorporate parts of the Program into other free
+programs whose distribution conditions are different, write to the author
+to ask for permission.  For software which is copyrighted by the Free
+Software Foundation, write to the Free Software Foundation; we sometimes
+make exceptions for this.  Our decision will be guided by the two goals
+of preserving the free status of all derivatives of our free software and
+of promoting the sharing and reuse of software generally.
+
+			    NO WARRANTY
+
+  11. BECAUSE THE PROGRAM IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
+FOR THE PROGRAM, TO THE EXTENT PERMITTED BY APPLICABLE LAW.  EXCEPT WHEN
+OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
+PROVIDE THE PROGRAM "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED
+OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.  THE ENTIRE RISK AS
+TO THE QUALITY AND PERFORMANCE OF THE PROGRAM IS WITH YOU.  SHOULD THE
+PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY SERVICING,
+REPAIR OR CORRECTION.
+
+  12. IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
+WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
+REDISTRIBUTE THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
+INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING
+OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED
+TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY
+YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER
+PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGES.
+
+		     END OF TERMS AND CONDITIONS
+
+	Appendix: How to Apply These Terms to Your New Programs
+
+  If you develop a new program, and you want it to be of the greatest
+possible use to the public, the best way to achieve this is to make it
+free software which everyone can redistribute and change under these terms.
+
+  To do so, attach the following notices to the program.  It is safest
+to attach them to the start of each source file to most effectively
+convey the exclusion of warranty; and each file should have at least
+the "copyright" line and a pointer to where the full notice is found.
+
+    <one line to give the program's name and a brief idea of what it does.>
+    Copyright (C) 19yy  <name of author>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+Also add information on how to contact you by electronic and paper mail.
+
+If the program is interactive, make it output a short notice like this
+when it starts in an interactive mode:
+
+    Gnomovision version 69, Copyright (C) 19yy name of author
+    Gnomovision comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
+    This is free software, and you are welcome to redistribute it
+    under certain conditions; type `show c' for details.
+
+The hypothetical commands `show w' and `show c' should show the appropriate
+parts of the General Public License.  Of course, the commands you use may
+be called something other than `show w' and `show c'; they could even be
+mouse-clicks or menu items--whatever suits your program.
+
+You should also get your employer (if you work as a programmer) or your
+school, if any, to sign a "copyright disclaimer" for the program, if
+necessary.  Here is a sample; alter the names:
+
+  Yoyodyne, Inc., hereby disclaims all copyright interest in the program
+  `Gnomovision' (which makes passes at compilers) written by James Hacker.
+
+  <signature of Ty Coon>, 1 April 1989
+  Ty Coon, President of Vice
+
+This General Public License does not permit incorporating your program into
+proprietary programs.  If your program is a subroutine library, you may
+consider it more useful to permit linking proprietary applications with the
+library.  If this is what you want to do, use the GNU Library General
+Public License instead of this License.
Index: PadicoTM/building-tools/COPYRIGHT
===================================================================
--- PadicoTM/building-tools/COPYRIGHT	(nonexistent)
+++ PadicoTM/building-tools/COPYRIGHT	(copie de travail)
@@ -0,0 +1,21 @@
+ Padico -- a High Performance Parallel and Distributed Computing Environment
+ Copyright (c) 2002-2010 INRIA and the University of Rennes 1
+ Alexandre DENIS <Alexandre.Denis@inria.fr>
+ Christian PEREZ <Christian.Perez@inria.fr>
+
+ The software has been registered at the Agency for the Protection of
+ Programs (APP) under the number IDDN.FR.001.260013.000.S.P.2002.000.10000.
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
Index: PadicoTM/building-tools/VERSION
===================================================================
--- PadicoTM/building-tools/VERSION	(nonexistent)
+++ PadicoTM/building-tools/VERSION	(copie de travail)
@@ -0,0 +1 @@
+0.5.0
Index: PadicoTM/building-tools/ax_check_compile_flag.m4
===================================================================
--- PadicoTM/building-tools/ax_check_compile_flag.m4	(nonexistent)
+++ PadicoTM/building-tools/ax_check_compile_flag.m4	(copie de travail)
@@ -0,0 +1,74 @@
+# ===========================================================================
+#   http://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_CHECK_COMPILE_FLAG(FLAG, [ACTION-SUCCESS], [ACTION-FAILURE], [EXTRA-FLAGS], [INPUT])
+#
+# DESCRIPTION
+#
+#   Check whether the given FLAG works with the current language's compiler
+#   or gives an error.  (Warnings, however, are ignored)
+#
+#   ACTION-SUCCESS/ACTION-FAILURE are shell commands to execute on
+#   success/failure.
+#
+#   If EXTRA-FLAGS is defined, it is added to the current language's default
+#   flags (e.g. CFLAGS) when the check is done.  The check is thus made with
+#   the flags: "CFLAGS EXTRA-FLAGS FLAG".  This can for example be used to
+#   force the compiler to issue an error when a bad flag is given.
+#
+#   INPUT gives an alternative input source to AC_COMPILE_IFELSE.
+#
+#   NOTE: Implementation based on AX_CFLAGS_GCC_OPTION. Please keep this
+#   macro in sync with AX_CHECK_{PREPROC,LINK}_FLAG.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Guido U. Draheim <guidod@gmx.de>
+#   Copyright (c) 2011 Maarten Bosmans <mkbosmans@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 4
+
+AC_DEFUN([AX_CHECK_COMPILE_FLAG],
+[AC_PREREQ(2.64)dnl for _AC_LANG_PREFIX and AS_VAR_IF
+AS_VAR_PUSHDEF([CACHEVAR],[ax_cv_check_[]_AC_LANG_ABBREV[]flags_$4_$1])dnl
+AC_CACHE_CHECK([whether _AC_LANG compiler accepts $1], CACHEVAR, [
+  ax_check_save_flags=$[]_AC_LANG_PREFIX[]FLAGS
+  _AC_LANG_PREFIX[]FLAGS="$[]_AC_LANG_PREFIX[]FLAGS $4 $1"
+  AC_COMPILE_IFELSE([m4_default([$5],[AC_LANG_PROGRAM()])],
+    [AS_VAR_SET(CACHEVAR,[yes])],
+    [AS_VAR_SET(CACHEVAR,[no])])
+  _AC_LANG_PREFIX[]FLAGS=$ax_check_save_flags])
+AS_VAR_IF(CACHEVAR,yes,
+  [m4_default([$2], :)],
+  [m4_default([$3], :)])
+AS_VAR_POPDEF([CACHEVAR])dnl
+])dnl AX_CHECK_COMPILE_FLAGS
Index: PadicoTM/building-tools/ax_mpi.m4
===================================================================
--- PadicoTM/building-tools/ax_mpi.m4	(nonexistent)
+++ PadicoTM/building-tools/ax_mpi.m4	(copie de travail)
@@ -0,0 +1,176 @@
+# ===========================================================================
+#          http://www.gnu.org/software/autoconf-archive/ax_mpi.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_MPI([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+#
+# DESCRIPTION
+#
+#   This macro tries to find out how to compile programs that use MPI
+#   (Message Passing Interface), a standard API for parallel process
+#   communication (see http://www-unix.mcs.anl.gov/mpi/)
+#
+#   On success, it sets the MPICC, MPICXX, MPIF77, or MPIFC output variable
+#   to the name of the MPI compiler, depending upon the current language.
+#   (This may just be $CC/$CXX/$F77/$FC, but is more often something like
+#   mpicc/mpiCC/mpif77/mpif90.) It also sets MPILIBS to any libraries that
+#   are needed for linking MPI (e.g. -lmpi or -lfmpi, if a special
+#   MPICC/MPICXX/MPIF77/MPIFC was not found).
+#
+#   Note that this macro should be used only if you just have a few source
+#   files that need to be compiled using MPI. In particular, you should
+#   neither overwrite CC/CXX/F77/FC with the values of
+#   MPICC/MPICXX/MPIF77/MPIFC, nor assume that you can use the same flags
+#   etc. as the standard compilers. If you want to compile a whole program
+#   using the MPI compiler commands, use one of the macros
+#   AX_PROG_{CC,CXX,FC}_MPI.
+#
+#   ACTION-IF-FOUND is a list of shell commands to run if an MPI library is
+#   found, and ACTION-IF-NOT-FOUND is a list of commands to run if it is not
+#   found. If ACTION-IF-FOUND is not specified, the default action will
+#   define HAVE_MPI.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Steven G. Johnson <stevenj@alum.mit.edu>
+#   Copyright (c) 2008 Julian C. Cummings <cummings@cacr.caltech.edu>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 8
+
+AU_ALIAS([ACX_MPI], [AX_MPI])
+AC_DEFUN([AX_MPI], [
+AC_PREREQ(2.50) dnl for AC_LANG_CASE
+
+AC_LANG_CASE([C], [
+	AC_REQUIRE([AC_PROG_CC])
+	AC_ARG_VAR(MPICC,[MPI C compiler command])
+	AC_CHECK_PROGS(MPICC, mpicc hcc mpxlc_r mpxlc mpcc cmpicc, $CC)
+	ax_mpi_save_CC="$CC"
+	CC="$MPICC"
+	AC_SUBST(MPICC)
+],
+[C++], [
+	AC_REQUIRE([AC_PROG_CXX])
+	AC_ARG_VAR(MPICXX,[MPI C++ compiler command])
+	AC_CHECK_PROGS(MPICXX, mpic++ mpicxx mpiCC hcp mpxlC_r mpxlC mpCC cmpic++, $CXX)
+	ax_mpi_save_CXX="$CXX"
+	CXX="$MPICXX"
+	AC_SUBST(MPICXX)
+],
+[Fortran 77], [
+	AC_REQUIRE([AC_PROG_F77])
+	AC_ARG_VAR(MPIF77,[MPI Fortran 77 compiler command])
+	AC_CHECK_PROGS(MPIF77, mpif77 hf77 mpxlf_r mpxlf mpf77 cmpifc, $F77)
+	ax_mpi_save_F77="$F77"
+	F77="$MPIF77"
+	AC_SUBST(MPIF77)
+],
+[Fortran], [
+	AC_REQUIRE([AC_PROG_FC])
+	AC_ARG_VAR(MPIFC,[MPI Fortran compiler command])
+	AC_CHECK_PROGS(MPIFC, mpif90 mpxlf95_r mpxlf90_r mpxlf95 mpxlf90 mpf90 cmpif90c, $FC)
+	ax_mpi_save_FC="$FC"
+	FC="$MPIFC"
+	AC_SUBST(MPIFC)
+])
+
+if test x = x"$MPILIBS"; then
+	AC_LANG_CASE([C], [AC_CHECK_FUNC(MPI_Init, [MPILIBS=" "])],
+		[C++], [AC_CHECK_FUNC(MPI_Init, [MPILIBS=" "])],
+		[Fortran 77], [AC_MSG_CHECKING([for MPI_Init])
+			AC_LINK_IFELSE([AC_LANG_PROGRAM([],[      call MPI_Init])],[MPILIBS=" "
+				AC_MSG_RESULT(yes)], [AC_MSG_RESULT(no)])],
+		[Fortran], [AC_MSG_CHECKING([for MPI_Init])
+			AC_LINK_IFELSE([AC_LANG_PROGRAM([],[      call MPI_Init])],[MPILIBS=" "
+				AC_MSG_RESULT(yes)], [AC_MSG_RESULT(no)])])
+fi
+AC_LANG_CASE([Fortran 77], [
+	if test x = x"$MPILIBS"; then
+		AC_CHECK_LIB(fmpi, MPI_Init, [MPILIBS="-lfmpi"])
+	fi
+	if test x = x"$MPILIBS"; then
+		AC_CHECK_LIB(fmpich, MPI_Init, [MPILIBS="-lfmpich"])
+	fi
+],
+[Fortran], [
+	if test x = x"$MPILIBS"; then
+		AC_CHECK_LIB(fmpi, MPI_Init, [MPILIBS="-lfmpi"])
+	fi
+	if test x = x"$MPILIBS"; then
+		AC_CHECK_LIB(mpichf90, MPI_Init, [MPILIBS="-lmpichf90"])
+	fi
+])
+if test x = x"$MPILIBS"; then
+	AC_CHECK_LIB(mpi, MPI_Init, [MPILIBS="-lmpi"])
+fi
+if test x = x"$MPILIBS"; then
+	AC_CHECK_LIB(mpich, MPI_Init, [MPILIBS="-lmpich"])
+fi
+
+dnl We have to use AC_TRY_COMPILE and not AC_CHECK_HEADER because the
+dnl latter uses $CPP, not $CC (which may be mpicc).
+AC_LANG_CASE([C], [if test x != x"$MPILIBS"; then
+	AC_MSG_CHECKING([for mpi.h])
+	AC_TRY_COMPILE([#include <mpi.h>],[],[AC_MSG_RESULT(yes)], [MPILIBS=""
+		AC_MSG_RESULT(no)])
+fi],
+[C++], [if test x != x"$MPILIBS"; then
+	AC_MSG_CHECKING([for mpi.h])
+	AC_TRY_COMPILE([#include <mpi.h>],[],[AC_MSG_RESULT(yes)], [MPILIBS=""
+		AC_MSG_RESULT(no)])
+fi],
+[Fortran 77], [if test x != x"$MPILIBS"; then
+	AC_MSG_CHECKING([for mpif.h])
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[      include 'mpif.h'])],[AC_MSG_RESULT(yes)], [MPILIBS=""
+		AC_MSG_RESULT(no)])
+fi],
+[Fortran], [if test x != x"$MPILIBS"; then
+	AC_MSG_CHECKING([for mpif.h])
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[      include 'mpif.h'])],[AC_MSG_RESULT(yes)], [MPILIBS=""
+		AC_MSG_RESULT(no)])
+fi])
+
+AC_LANG_CASE([C], [CC="$ax_mpi_save_CC"],
+	[C++], [CXX="$ax_mpi_save_CXX"],
+	[Fortran 77], [F77="$ax_mpi_save_F77"],
+	[Fortran], [FC="$ax_mpi_save_FC"])
+
+AC_SUBST(MPILIBS)
+
+# Finally, execute ACTION-IF-FOUND/ACTION-IF-NOT-FOUND:
+if test x = x"$MPILIBS"; then
+        $2
+        :
+else
+        ifelse([$1],,[AC_DEFINE(HAVE_MPI,1,[Define if you have the MPI library.])],[$1])
+        :
+fi
+])dnl AX_MPI
Index: PadicoTM/building-tools/ax_openmp.m4
===================================================================
--- PadicoTM/building-tools/ax_openmp.m4	(nonexistent)
+++ PadicoTM/building-tools/ax_openmp.m4	(copie de travail)
@@ -0,0 +1,119 @@
+# ===========================================================================
+#         http://www.gnu.org/software/autoconf-archive/ax_openmp.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_OPENMP([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+#
+# DESCRIPTION
+#
+#   This macro tries to find out how to compile programs that use OpenMP a
+#   standard API and set of compiler directives for parallel programming
+#   (see http://www-unix.mcs/)
+#
+#   On success, it sets the OPENMP_CFLAGS/OPENMP_CXXFLAGS/OPENMP_F77FLAGS
+#   output variable to the flag (e.g. -omp) used both to compile *and* link
+#   OpenMP programs in the current language.
+#
+#   NOTE: You are assumed to not only compile your program with these flags,
+#   but also link it with them as well.
+#
+#   If you want to compile everything with OpenMP, you should set:
+#
+#     CFLAGS="$CFLAGS $OPENMP_CFLAGS"
+#     #OR#  CXXFLAGS="$CXXFLAGS $OPENMP_CXXFLAGS"
+#     #OR#  FFLAGS="$FFLAGS $OPENMP_FFLAGS"
+#
+#   (depending on the selected language).
+#
+#   The user can override the default choice by setting the corresponding
+#   environment variable (e.g. OPENMP_CFLAGS).
+#
+#   ACTION-IF-FOUND is a list of shell commands to run if an OpenMP flag is
+#   found, and ACTION-IF-NOT-FOUND is a list of commands to run it if it is
+#   not found. If ACTION-IF-FOUND is not specified, the default action will
+#   define HAVE_OPENMP.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Steven G. Johnson <stevenj@alum.mit.edu>
+#   Copyright (c) 2015 John W. Peterson <jwpeterson@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 11
+
+AC_DEFUN([AX_OPENMP], [
+AC_PREREQ([2.69]) dnl for _AC_LANG_PREFIX
+
+AC_CACHE_CHECK([for OpenMP flag of _AC_LANG compiler], ax_cv_[]_AC_LANG_ABBREV[]_openmp, [save[]_AC_LANG_PREFIX[]FLAGS=$[]_AC_LANG_PREFIX[]FLAGS
+ax_cv_[]_AC_LANG_ABBREV[]_openmp=unknown
+# Flags to try:  -fopenmp (gcc), -openmp (icc), -mp (SGI & PGI),
+#                -xopenmp (Sun), -omp (Tru64), -qsmp=omp (AIX), none
+ax_openmp_flags="-fopenmp -openmp -mp -xopenmp -omp -qsmp=omp none"
+if test "x$OPENMP_[]_AC_LANG_PREFIX[]FLAGS" != x; then
+  ax_openmp_flags="$OPENMP_[]_AC_LANG_PREFIX[]FLAGS $ax_openmp_flags"
+fi
+for ax_openmp_flag in $ax_openmp_flags; do
+  case $ax_openmp_flag in
+    none) []_AC_LANG_PREFIX[]FLAGS=$save[]_AC_LANG_PREFIX[] ;;
+    *) []_AC_LANG_PREFIX[]FLAGS="$save[]_AC_LANG_PREFIX[]FLAGS $ax_openmp_flag" ;;
+  esac
+  AC_LINK_IFELSE([AC_LANG_SOURCE([[
+@%:@include <omp.h>
+
+static void
+parallel_fill(int * data, int n)
+{
+  int i;
+@%:@pragma omp parallel for
+  for (i = 0; i < n; ++i)
+    data[i] = i;
+}
+
+int
+main()
+{
+  int arr[100000];
+  omp_set_num_threads(2);
+  parallel_fill(arr, 100000);
+  return 0;
+}
+]])],[ax_cv_[]_AC_LANG_ABBREV[]_openmp=$ax_openmp_flag; break],[])
+done
+[]_AC_LANG_PREFIX[]FLAGS=$save[]_AC_LANG_PREFIX[]FLAGS
+])
+if test "x$ax_cv_[]_AC_LANG_ABBREV[]_openmp" = "xunknown"; then
+  m4_default([$2],:)
+else
+  if test "x$ax_cv_[]_AC_LANG_ABBREV[]_openmp" != "xnone"; then
+    OPENMP_[]_AC_LANG_PREFIX[]FLAGS=$ax_cv_[]_AC_LANG_ABBREV[]_openmp
+  fi
+  m4_default([$1], [AC_DEFINE(HAVE_OPENMP,1,[Define if OpenMP is enabled])])
+fi
+])dnl AX_OPENMP
Index: PadicoTM/building-tools/common_rules.mk.in
===================================================================
--- PadicoTM/building-tools/common_rules.mk.in	(nonexistent)
+++ PadicoTM/building-tools/common_rules.mk.in	(copie de travail)
@@ -0,0 +1,218 @@
+
+# common_rules.mk: begin
+# @configure_input@
+
+.SUFFIXES:
+
+.PHONY: all install install-lib install-libexec install-bin install-include clean clean-obj clean-doxygen doxygen distclean subdirs modules dep $(SUBDIRS) $(TARGET_MOD) $(MODULES)
+
+all: $(TARGETS)
+
+subdirs: $(SUBDIRS)
+
+$(SUBDIRS):
+	$(SHOWMK) "  [SUBDIR] make '$(if $(MAKECMDGOALS),$(MAKECMDGOALS),all)' in: $@"
+	$(Q)$(MAKE) -C $@ $(MAKECMDGOALS)
+
+# ## docs, only if doxyfile exists
+ifneq ($(wildcard doc/Doxyfile),)
+doxygen: @abs_top_builddir@/doc/Doxyfile @srcdir@/README.md
+	$(SHOWMK) "  [DOXYGEN] $(MYDIR)/"
+	$(Q)doxygen doc/Doxyfile
+	@echo
+	@echo "# documentation generated in: @top_builddir@/doc/html/index.html"
+	@echo
+
+clean-doxygen:
+	$(SHOWMK) "  [CLEAN]  $(MYDIR)/doc/html"
+	$(Q)$(RM) doc/html
+
+TARGET_CLEAN += clean-doxygen
+
+upload-doc: doxygen
+	@echo "  [RSYNC]  $(package_short_name)"
+	rsync -rcvz @abs_top_builddir@/doc/html/ scm.gforge.inria.fr:/home/groups/pm2/htdocs/$(package_short_name)/doc/
+endif
+
+clean: subdirs clean-obj clean-mod $(TARGET_CLEAN)
+	$(SHOWMK) "  [CLEAN]  $(MYDIR)/"
+
+clean-obj: subdirs
+	$(Q)$(RM) *.o *.pic.o *.so *.dylib *.d .*.d .*.stamp $(TARGET_BIN) $(TARGET_LIBEXEC) $(TARGET_LIB) $(CLEAN_MORE)
+
+clean-mod: subdirs
+	$(Q)(  \
+          for mod in $(patsubst %/meta.xml, %, $(wildcard */meta.xml)); do  \
+	    $(ECHO) "  [CLEAN]  $(MYDIR)/$${mod}" ; \
+	    if [ -r $${mod}/meta.xml ]; then rm -r $${mod}; fi ; \
+	  done  \
+	)
+
+distclean: clean
+	$(SHOWMK) "  [DISTCLEAN] $(MYDIR)/"
+	$(Q)$(RM) config.log config.status $(PADICO_OUT_FILES)
+
+install: install-dir install-subdirs $(TARGETS)
+
+install-dir: install-sbin install-lib install-mod install-libexec install-bin install-include install-etc install-init install-pkgconfig
+
+install-subdirs: subdirs 
+
+install-lib: $(INSTALL_LIB)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in lib/:     $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_LIB_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_DATA) $^ $(INSTALL_LIB_PATH)
+
+install-mod: $(INSTALL_MOD)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in lib/padico/: $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_MOD_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(CP) -r $^ $(INSTALL_MOD_PATH)/
+
+install-libexec: $(INSTALL_LIBEXEC)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in libexec/: $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d  $(INSTALL_LIBEXEC_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_PROG) $^ $(INSTALL_LIBEXEC_PATH)
+
+install-bin: $(INSTALL_BIN)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in bin/:     $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_BIN_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_PROG) $^ $(INSTALL_BIN_PATH)
+
+install-include: $(INSTALL_INCLUDE)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in include/: $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_INCLUDE_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_DATA) $^ $(INSTALL_INCLUDE_PATH)
+
+install-etc: $(INSTALL_ETC)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in etc/:     $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_ETC_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_DATA) -m 644 $^ $(INSTALL_ETC_PATH)
+
+install-init: $(INSTALL_INIT)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in etc/init.d/: $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_ETC_PATH)/init.d/
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_DATA) -m 644 $^ $(INSTALL_ETC_PATH)/init.d/
+
+install-sbin: $(INSTALL_SBIN)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in sbin/:    $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_SBIN_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_PROG) $^ $(INSTALL_SBIN_PATH)
+
+install-pkgconfig: $(INSTALL_PKGCONFIG)
+	@$(if $^,,$(IGNORE)) $(SHOWMK) "  [INST]   in lib/pkgconfig/: $(^F)"
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL) -d $(INSTALL_PKGCONFIG_PATH)
+	@$(if $^,,$(IGNORE)) $(Q)$(INSTALL_DATA) $^ $(INSTALL_PKGCONFIG_PATH)
+
+ifeq (@HAVE_DYNLIB@,yes)
+@DYNLIB_PREFIX@%@DYNLIB_EXT@:
+	@$(SHOWMK) "  [LIB-LD] $@"
+	$(Q)$(DYNLIB_LD) $(LDFLAGS) -o $@ $+ $(LIBS)
+endif
+
+lib%.a:
+	@$(SHOWMK) "  [AR]     $@"
+	@$(AR) cr $@ $^
+	@$(SHOWMK) "  [RANLIB] $@"
+	@$(RANLIB) $@
+
+$(TARGET_BIN):
+	@$(SHOWMK) "  [LD]     $@"
+	$(Q)$(LD) $(LDFLAGS) -o $@ $+ $(LIBS) $(BINLIBS)
+
+$(TARGET_LIBEXEC):
+	@$(SHOWMK) "  [LD]     $@"
+	$(Q)$(LD) $(LDFLAGS) -o $@ $+ $(LIBS)
+
+$(TARGET_MOD):
+	$(Q)$(MAKE) --silent .$@.stamp _MODOBJS='$(filter %o %.pic %.so %.a,$+)' _MODLIBS='$(MODLIBS)'
+
+.%.stamp: $(_MODOBJS)
+	@$(SHOWMK) "  [MKMOD]  $* [$(notdir $(_MODOBJS))]"
+	$(Q)$(MKMOD) -o $* $(_MODOBJS) $(_MODLIBS)
+	@$(TOUCH) $@
+
+# ## Rule to make modules
+# build module, then install it, before building next
+
+$(MODULES):
+	$(SHOWMK) "  [MODULE] make '$(if $(MAKECMDGOALS),$(MAKECMDGOALS),all)' in: $@"
+	$(Q)$(MAKE) -C $@ all
+	$(Q)$(MAKE) -C $@ install
+
+# don't need .d files for cleanning!
+ifeq ($(MAKECMDGOALS),clean)
+  PADICO_USE_MKDEPEND = no
+endif
+
+ifeq ($(PADICO_USE_MKDEPEND),yes)
+# rules with dependancy checking
+
+.PRECIOUS: %.d
+
+%.pic.o: %.c %.d
+	@$(SHOWMK) "  [CC]     $@"
+	$(Q)$(CC) $(CFLAGS) $(CPICFLAGS) -c -o $@ $<
+
+%.o: %.c %.d
+	@$(SHOWMK) "  [CC]     $@"
+	$(Q)$(CC) $(CFLAGS) -c -o $@ $<
+
+%.d: %.c $(PADICO_CONFIG_FILES)
+	@$(SHOWMK) "  [CDEP]   $@"
+	$(Q)$(MKDIR) $(dir $@)
+	$(Q)$(CC_DEP) $(CFLAGS) -DMKDEPEND $< > $@
+	$(Q)$(ECHO) >> $@
+	$(Q)$(CC_DEP) -MT $*.pic.o $(CFLAGS) -DMKDEPEND $< >> $@
+
+TARGET_DEP += $(patsubst $(srcdir)/%.c,  ./%.d, $(wildcard $(srcdir)/*.c))
+
+ifeq ($(HAVE_CXX),yes)
+%.pic.o: %.cc %.d
+	@$(SHOWMK) "  [C++]    $@"
+	$(Q)$(CXX) $(CXXFLAGS) $(CXXPICFLAGS) -c -o $@ $<
+
+%.o: %.cc %.d
+	@$(SHOWMK) "  [C++]    $@"
+	$(Q)$(CXX) $(CXXFLAGS) -c -o $@ $<
+
+%.d: %.cc $(PADICO_CONFIG_FILES)
+	@$(SHOWMK) "  [C++DEP] $@"
+	$(Q)$(CXX_DEP) $(CXXFLAGS) $< > $@
+
+TARGET_DEP += $(patsubst $(srcdir)/%.cc, ./%.d, $(wildcard $(srcdir)/*.cc))
+endif
+
+
+CLEAN_MORE += $(TARGET_DEP)
+
+dep: $(TARGET_DEP)
+
+_DEP_FILES = $(wildcard *.d)
+
+ifneq ($(strip $(_DEP_FILES)),)
+-include $(_DEP_FILES)
+endif
+
+else
+# PADICO_USE_MKDEPEND = no
+%.o: %.c
+	@$(SHOWMK) "  [CC]     $@"
+	$(Q)$(CC) $(CFLAGS) -c -o $@ $<
+
+%.pic.o: %.c
+	@$(SHOWMK) "  [CC]     $@"
+	$(Q)$(CC) $(CFLAGS) $(CPICFLAGS) -c -o $@ $<
+
+ifeq ($(HAVE_CXX),yes)
+%.o: %.cc
+	@$(SHOWMK) "  [C++]    $@"
+	$(Q)$(CXX) $(CXXFLAGS) -c -o $@ $<
+
+%.pic.o: %.cc
+	@$(SHOWMK) "  [C++]    $@"
+	$(Q)$(CXX) $(CXXFLAGS) $(CXXPICFLAGS) -c -o $@ $<
+endif
+endif
+# end: PADICO_USE_MKDEPEND
+
+# common_rules.mk: end
Index: PadicoTM/building-tools/common_vars.mk.in
===================================================================
--- PadicoTM/building-tools/common_vars.mk.in	(nonexistent)
+++ PadicoTM/building-tools/common_vars.mk.in	(copie de travail)
@@ -0,0 +1,126 @@
+
+# common_vars.mk: begin
+# @configure_input@
+
+padico_builddir := @padico_builddir@
+padico_srcdir   := @padico_srcdir@
+prefix          := @prefix@
+
+package_name       := @package_name@
+package_short_name := @package_short_name@
+
+# path used for install; may be overriden in makefiles
+INSTALL_PREFIX         = $(prefix)
+INSTALL_BIN_PATH       = $(INSTALL_PREFIX)/bin
+INSTALL_SBIN_PATH      = $(INSTALL_PREFIX)/sbin
+INSTALL_LIB_PATH       = $(INSTALL_PREFIX)/lib
+INSTALL_ETC_PATH       = $(INSTALL_PREFIX)/etc
+INSTALL_LIBEXEC_PATH   = $(INSTALL_PREFIX)/libexec
+INSTALL_INCLUDE_PATH   = $(INSTALL_PREFIX)/include
+INSTALL_PKGCONFIG_PATH = $(INSTALL_LIB_PATH)/pkgconfig
+INSTALL_MOD_PATH       = $(INSTALL_LIB_PATH)/padico
+
+# ## Various flags
+# preprocessor options - usually "-D_REENTRANT"
+CPPFLAGS        = @CPPFLAGS@
+# common flags for C code - usually "-Wall"
+BASE_CFLAGS    := @BASE_CFLAGS@
+# flags for debug flavors - usually "-g"
+DEBUG_CFLAGS   := -DDEBUG @DEBUG_CFLAGS@
+# flags for optimize - usually "-O2" or "-O4 -mcpu=athlon" and the likes
+OPT_CFLAGS     := -DNDEBUG @OPT_CFLAGS@
+# flags for producing Position Independent Code
+CPICFLAGS      := @CPICFLAGS@
+# is C++ compiler used
+HAVE_CXX       := @HAVE_CXX@
+# common flags for C++
+BASE_CXXFLAGS  := @BASE_CXXFLAGS@
+# debug for C++
+DEBUG_CXXFLAGS := -DDEBUG @DEBUG_CXXFLAGS@
+# optimize for C++
+OPT_CXXFLAGS   := -DNDEBUG @OPT_CXXFLAGS@
+# PIC flags for C++
+CXXPICFLAGS    := @CXXPICFLAGS@
+# optimize linker hash tables
+OPT_LDFLAGS    := @OPT_LDFLAGS@
+
+CFLAGS   = $(CPPFLAGS) $(BASE_CFLAGS)   -I. -I@includedir@
+CXXFLAGS = $(CPPFLAGS) $(BASE_CXXFLAGS) -I. -I@includedir@
+
+# basic shell commands
+ECHO    = echo
+RM      = rm -rf
+CP      = cp -f
+LN      = ln -fs
+MKDIR   = mkdir -p
+INSTALL = install -p
+SHELL   = /bin/sh
+TOUCH   = touch
+IGNORE  = @BIN_TRUE@
+INSTALL_DATA = $(INSTALL) -m 644
+INSTALL_PROG = $(INSTALL) -m 755
+
+# build & link
+LDFLAGS    = @LDFLAGS@
+LIBS       = @LIBS@
+BINLIBS    =
+CC         = @CC@
+CXX        = @CXX@
+LD         = @LD@
+CC_DEP     = @CC_DEP@
+CXX_DEP    = @CXX_DEP@
+DYNLIB_LD  = @DYNLIB_LD@
+DYNLIB_EXT = @DYNLIB_EXT@
+DSO_LD     = @DSO_LD@
+DSO_EXT    = @DSO_EXT@
+AR         = ar
+RANLIB     = ranlib
+
+# control verbosity
+ifeq ($(VERBOSE),yes)
+# verbose
+Q         := 
+SHOWMK    := @$(IGNORE)
+MKMOD     := @puk_root@/bin/padico-mkmod
+INSTALL   := $(INSTALL) -v
+else
+# quiet
+Q         := @
+SHOWMK    := @$(ECHO)
+MAKEFLAGS += --no-print-directory
+MKMOD     := @puk_root@/bin/padico-mkmod --quiet
+endif
+
+# Default choices for optimize/debug follows PadicoTM configure
+ifeq ($(ENABLE_OPTIMIZE),)
+ENABLE_OPTIMIZE = @ENABLE_OPTIMIZE_FLAG@
+endif
+ifeq ($(ENABLE_DEBUG),)
+ENABLE_DEBUG = @ENABLE_DEBUG_FLAG@
+endif
+# set the appropriate flags for optimize/debug
+ifeq ($(ENABLE_OPTIMIZE),yes)
+CFLAGS    += $(OPT_CFLAGS)
+CXX_FLAGS += $(OPT_CXXFLAGS)
+LDFLAGS   += $(OPT_CFLAGS) $(OPT_LDFLAGS)
+endif
+ifeq ($(ENABLE_DEBUG),yes)
+CFLAGS    += $(DEBUG_CFLAGS)
+CXX_FLAGS += $(DEBUG_CXXFLAGS)
+endif
+
+MODLIBS = -DENABLE_OPTIMIZE=$(ENABLE_OPTIMIZE) -DENABLE_DEBUG=$(ENABLE_DEBUG)
+
+# config files that trigger rebuild if changed
+PADICO_CONFIG_FILES := $(wildcard Makefile config.status common_vars.mk common_rules.mk)
+
+# out files generated by configure; will be remove by distclean
+PADICO_OUT_FILES := @PADICO_OUT_FILES@
+
+# always use mkdepend, except for clean
+PADICO_USE_MKDEPEND = yes
+
+# directory used for pretty-print
+MYDIR := $(notdir $(CURDIR))
+
+# common_vars.mk: end
Index: PadicoTM/building-tools/compiler.m4
===================================================================
--- PadicoTM/building-tools/compiler.m4	(nonexistent)
+++ PadicoTM/building-tools/compiler.m4	(copie de travail)
@@ -0,0 +1,229 @@
+dnl -*- mode: Autoconf;-*-
+
+m4_include([ax_check_compile_flag.m4])
+
+AC_DEFUN([AC_PADICO_CXX],
+  [
+    MYCXXFLAGS="$CXXFLAGS"
+    AC_PROG_CXX
+    AC_PROG_CXXCPP
+    case `uname -s` in
+      SunOS)
+        case ${CXX} in
+          c++|g++*) CXXFLAGS="-g"
+    		    CXXPICFLAGS="-fPIC"
+               	    CXXDEPFLAGS="-MM";;
+               CC)  CXXFLAGS="-g"
+		    CXXPICFLAGS="-Kpic"
+	            CXXDEPFLAGS="-xM1"
+		    CXXSHFLAGS="-G";;
+        esac
+      ;;
+      Linux|*BSD)
+        BASE_CXXFLAGS=""
+        DEBUG_CXXFLAGS="-g"
+        OPT_CXXFLAGS="-O2"
+        CXXPICFLAGS="-fPIC"
+        CXXDEPFLAGS="-MM"
+      ;;
+      IRIX64)
+        CXXFLAGS="-Wall -Dirix"
+        CXXPICFLAGS="-fPIC"
+        CXXDEPFLAGS="-MM"
+        CXXDEBUGFLAGS="-g"
+        CXXOPTFLAGS="-O2" dnl it's *really* -O2, not -O4. Else, it doesn't work with omniORB 3
+      ;;
+      Darwin)
+        CXXFLAGS="-g -Wall -traditional-cpp"
+        CXXPICFLAGS="-fPIC"
+        CXXDEPFLAGS="-MM"
+      ;;
+      *) # What is this f$*k!@ OS? :-)
+        AC_MSG_ERROR([Unknown OS/platform: `uname -s`])
+      ;;
+    esac
+    BASE_CXXFLAGS="$MYCXXFLAGS $BASE_CXXFLAGS"
+    AC_SUBST(CXXPICFLAGS)
+    AC_SUBST(BASE_CXXFLAGS)
+    AC_SUBST(DEBUG_CXXFLAGS)
+    AC_SUBST(OPT_CXXFLAGS)
+    HAVE_CXX=yes
+    AC_SUBST([HAVE_CXX])
+  ])
+
+AC_DEFUN([AC_PADICO_SAVEFLAGS],
+  [
+    MYCPPFLAGS="$CPPFLAGS"
+    MYCFLAGS="$CFLAGS"
+    MYLDFLAGS="$LDFLAGS"
+  ])
+  
+AC_DEFUN([AC_PADICO_COMPILER],
+  [
+    dnl - save current state
+    AC_REQUIRE([AC_PADICO_SAVEFLAGS])
+    AC_REQUIRE([AC_PROG_CC])
+    AC_REQUIRE([AC_PROG_CPP])
+    AC_REQUIRE([AC_PROG_CC_C99])
+    dnl --- compute flags depending on the OS type
+    case `uname -s` in
+      SunOS)
+        binary_format=elf
+        dnl - dynamic libraries
+        LDFLAGS=
+        SYMBOL_PREFIX=""
+        DYNLIB_PREFIX="lib"
+        DYNLIB_EXT=".so"
+        DYNLIB_FLAGS="-shared"
+        DSO_EXT=".so"
+        DSO_FLAGS="-shared"
+        case ${CC} in
+          gcc*) CFLAGS="-g -Wall"
+                CPICFLAGS="-fPIC"
+	        CDEPFLAGS="-MM";;
+          cc)   CFLAGS="-g"
+  	        CPICFLAGS="-Kpic"
+	        CDEPFLAGS="-xM1";;
+        esac
+      ;;
+      Linux|*BSD)
+        binary_format=elf
+        LDFLAGS="-rdynamic"
+        SYMBOL_PREFIX=""
+        DYNLIB_PREFIX="lib"
+        DYNLIB_EXT=".so"
+        DYNLIB_FLAGS="-shared -rdynamic"
+        DSO_EXT=".so"
+        DSO_FLAGS="-shared -rdynamic"
+        OPT_LDFLAGS="-Wl,-O1"
+        dnl gcc
+        BASE_CFLAGS="-Wall -pipe"
+        DEBUG_CFLAGS="-g"
+        AX_CHECK_COMPILE_FLAG([-xHOST], [OPT_CFLAGS_NATIVE="-xHOST"], [])
+	if test "x${OPT_CFLAGS_NATIVE}" = "x"; then
+          AX_CHECK_COMPILE_FLAG([-march=native], [OPT_CFLAGS_NATIVE="-march=native"], [])
+	fi
+        AX_CHECK_COMPILE_FLAG([-ipo], [OPT_CFLAGS_LTO="-ipo"], [])
+	if test "x${OPT_CFLAGS_LTO}" = "x"; then
+	  AX_CHECK_COMPILE_FLAG([-flto], [OPT_CFLAGS_LTO="-flto"], [])
+	fi
+	OPT_CFLAGS="-O3 $OPT_CFLAGS_NATIVE $OPT_CFLAGS_LTO"
+        CPICFLAGS="-fPIC"
+        CDEPFLAGS="-MM"
+      ;;
+      IRIX64)
+        binary_format=elf
+        dnl - dynamic libraries
+        LDFLAGS="-rdynamic"
+        SYMBOL_PREFIX="" 
+        DYNLIB_PREFIX="lib"
+        DYNLIB_EXT=".so"
+        DYNLIB_FLAGS="-shared"
+        DSO_EXT=".so"
+        DSO_FLAGS="-shared"
+        dnl gcc-specific flags... but no other compiler supported
+        CFLAGS="-Wall -Dirix"
+        CPICFLAGS="-fPIC"
+        CDEPFLAGS="-MM"
+        CDEBUGFLAGS="-g"
+        COPTFLAGS="-O4"
+      ;;
+      Darwin)
+        binary_format="Mach-O"
+        dnl - dynamic libraries
+        LDFLAGS=""
+        SYMBOL_PREFIX="_"
+        DYNLIB_PREFIX="lib"
+        DYNLIB_EXT=".dylib"
+        DYNLIB_FLAGS="-flat_namespace -undefined suppress -fno-common -dynamiclib"
+        DSO_EXT=""
+        DSO_FLAGS="-flat_namespace -undefined suppress -bundle"
+        dnl - gcc
+        CFLAGS="-g -Wall -traditional-cpp -fno-common -Wno-long-double"
+        CPICFLAGS="-fPIC"
+        CDEPFLAGS="-MM"
+      ;;
+      *) # What is this f$*k!@ OS? :-)
+        AC_MSG_ERROR([Unknown OS/platform: `uname -s`])
+      ;;
+    esac
+    dnl - user specified and standard flags
+    CPPFLAGS="$MYCPPFGLAGS -D_REENTRANT -D_GNU_SOURCE"
+    LDFLAGS="$MYLDFLAGS $LDFLAGS"
+    BASE_CFLAGS="$MYCFLAGS $BASE_CFLAGS"
+    AC_SUBST(CPPFLAGS)
+    AC_SUBST(CDEPFLAGS)
+    AC_SUBST(CPICFLAGS)
+    AC_SUBST(DEBUG_CFLAGS)
+    AC_SUBST(BASE_CFLAGS)
+    AC_SUBST(OPT_CFLAGS)
+    AC_SUBST(OPT_LDFLAGS)
+    AC_SUBST(binary_format)
+  ])
+
+
+AC_DEFUN([AC_PADICO_GCC_VERSION],
+	 [ AC_REQUIRE([AC_PADICO_VERSION])
+	   AC_REQUIRE([AC_PADICO_COMPILER])
+	   AC_MSG_CHECKING([for gcc version])
+	   if test "x$GCC" != "xyes"; then
+	     AC_MSG_ERROR([ $PACKAGE_STRING requires GNU C compiler.])
+	   else
+	     gcc_version="`${CC} -dumpversion`"
+	     case $gcc_version in
+	       2.9*|3.0.4|3.1*|3.2*|3.3*|3.4*|4.*|5.*|6.*|7.*|7)
+	         AC_MSG_RESULT([ ${gcc_version} ok])
+	       ;;
+	       2.7*|2.8*|3.0*|2.96*)
+	         AC_MSG_RESULT([ ${gcc_version} not supported-- warning: gcc version not supported])
+	       ;;
+	       *)
+	         AC_MSG_RESULT([ ${gcc_version} unknown-- report how it works!])
+	       ;;
+	     esac
+	   fi
+	  ])
+
+AC_DEFUN([AC_PADICO_GXX_VERSION],
+	 [ AC_REQUIRE([AC_PADICO_VERSION])
+	   AC_REQUIRE([AC_PADICO_CXX])
+           AC_MSG_CHECKING([for g++ version])
+	   if test "x$GXX" != "xyes"; then
+	     AC_MSG_ERROR([ $PACKAGE_STRING requires GNU C++ compiler.])
+	   else
+	     gxx_version="`${CXX} -dumpversion`"
+	     case $gxx_version in
+	       2.9*|3.0.4|3.1*|3.2*|3.3*|3.4*|4.*|5.*|6.*|7.*|7)
+	         AC_MSG_RESULT([ ${gxx_version} ok])
+	       ;;
+	       2.*|3.0*)
+	         AC_MSG_RESULT([ ${gxx_version} --warning: g++ version not supported. No CORBA support])
+	       ;;
+	       *)
+	         AC_MSG_RESULT([ ${gxx_version} unkown --report how it works!])
+	       ;;
+	     esac
+	   fi
+         ])
+
+# ## dynamic objects
+AC_DEFUN([AC_PADICO_DSO],
+	 [
+	   AC_REQUIRE([AC_PADICO_COMPILER])
+	   AC_MSG_CHECKING([for dynamic object (DSO) linker])
+	   DSO_LD="${CC} ${CPICFLAGS} ${DSO_FLAGS}"
+	   AC_MSG_RESULT([${DSO_LD}])
+	   AC_SUBST(DSO_LD)
+
+	   AC_MSG_CHECKING([for DSO extension])
+	   AC_MSG_RESULT([${DSO_EXT}])
+	   AC_SUBST(DSO_EXT)
+	   AC_DEFINE_UNQUOTED(DSO_EXT, "${DSO_EXT}",
+	     [Extension for bundles])
+
+	   AC_MSG_CHECKING([for symbol prefix])
+	   AC_MSG_RESULT([${SYMBOL_PREFIX}])
+	   AC_SUBST(SYMBOL_PREFIX)
+	   AC_DEFINE_UNQUOTED(SYMBOL_PREFIX, "${SYMBOL_PREFIX}",
+	     [prefix prepended to symbols by the ABI (usually "_")])
+	 ])
Index: PadicoTM/building-tools/features.m4
===================================================================
--- PadicoTM/building-tools/features.m4	(nonexistent)
+++ PadicoTM/building-tools/features.m4	(copie de travail)
@@ -0,0 +1,321 @@
+dnl -*- mode: Autoconf;-*-
+
+dnl -- Optionnal features
+dnl --------------------------------------------------------
+
+AC_DEFUN([AC_PADICO_CHECK_ZLIB],
+	 [
+	   AC_CHECK_HEADER(zlib.h, 
+ 	        [ HAVE_ZLIB_H="yes"
+	          HAVE_ZLIB="yes"
+	          AC_DEFINE(HAVE_ZLIB_H, 1, [Define if you have the <zlib.h> header file.])],
+	        [ HAVE_ZLIB_H="no"
+	          HAVE_ZLIB="no" ])
+	   AC_SUBST(HAVE_ZLIB)
+	 ])
+
+AC_DEFUN([AC_PADICO_CHECK_BZLIB],
+	 [
+	   AC_CHECK_HEADER(bzlib.h, 
+	        [ HAVE_BZLIB_H="yes"
+	          HAVE_BZLIB="yes"
+	          AC_DEFINE(HAVE_BZLIB_H, 1, [Define if you have the <bzlib.h> header file.])],
+	        [ HAVE_BZLIB_H="no"
+	          HAVE_BZLIB="no" ])
+	   AC_SUBST(HAVE_BZLIB)
+ 	 ])
+
+AC_DEFUN([AC_PADICO_CHECK_LZ4],
+	 [
+	   AC_CHECK_HEADER(lz4.h, 
+	        [ HAVE_LZ4_H="yes"
+	          HAVE_LIBLZ4="yes"
+	          AC_DEFINE(HAVE_LZ4_H, 1, [Define if you have the <lz4.h> header file.])],
+	        [ HAVE_LZ4_H="no"
+	          HAVE_LIBLZ4="no" ])
+  	   AC_SUBST(HAVE_LIBLZ4)
+	 ])
+
+# ## checks for hwloc
+# input: need_hwloc=yes|""
+# output: with_hwloc, requires_hwloc, hwloc_CFLAGS|LIBS
+AC_DEFUN([AC_PADICO_CHECK_HWLOC],
+	 [
+	   AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+	   AC_ARG_WITH([hwloc],
+	               [AS_HELP_STRING([--with-hwloc], [use hwloc @<:@default=auto@:>@] )])
+           if test "x${need_hwloc}" = "xyes"; then
+  	     with_hwloc=yes
+	   fi
+	   HAVE_HWLOC=no
+	   if test "x${with_hwloc}" != "xno"; then
+ 	     PKG_CHECK_MODULES([hwloc], [ hwloc ], [HAVE_HWLOC=yes], [HAVE_HWLOC=no])
+	   fi
+	   if test "x${with_hwloc}" = "xyes" -a "${HAVE_HWLOC}" = "no"; then
+	     AC_MSG_ERROR([ required library hwloc not found.])
+ 	   fi
+	   if test "${HAVE_HWLOC}" = "yes"; then
+	     with_hwloc=yes
+	     requires_hwloc=hwloc
+	     AC_DEFINE([HAVE_HWLOC], [1], [whether hwloc is present])
+	   fi
+	   AC_SUBST([requires_hwloc])
+	   AC_SUBST([with_hwloc])
+	 ])
+
+
+dnl -- check for Infiniband (ibverbs)
+dnl --
+AC_DEFUN([AC_PADICO_CHECK_INFINIBAND],
+  [
+    HAVE_INFINIBAND=no
+    with_ibverbs=check
+    AC_ARG_WITH([ibverbs],
+      [AS_HELP_STRING([--with-ibverbs], [use Infiniband ibverbs @<:@default=check@:>@] )])
+    if test "x${with_ibverbs}" != "xno"; then
+      save_CPPFLAGS="${CPPFLAGS}"
+      save_LIBS="${LIBS}"
+      if test "x${with_ibverbs}" != "xcheck" -a "${with_ibverbs}" != "xyes"; then
+        IBHOME="${with_ibverbs}"
+      fi
+      if test "x$IBHOME" != "x" -a "x${IBHOME}" != "x/usr"; then
+    	ibverbs_CPPLAGS="-I${IBHOME}/include"
+        ibverbs_LIBS="-Wl,-rpath,${IBHOME}/lib -L${IBHOME}/lib"
+      fi
+      CPPFLAGS="${CPPFLAGS} ${ibverbs_CPPFLAGS}"
+      LIBS="${LIBS} ${ibverbs_LIBS}"
+      AC_CHECK_HEADER(infiniband/verbs.h, [ HAVE_INFINIBAND_H=yes ], [ HAVE_INFINIBAND_H=no  ])
+      AC_CHECK_LIB(ibverbs, ibv_open_device, [ HAVE_LIBIBVERBS=yes ], [ HAVE_LIBIBVERBS=no ])
+      if test ${HAVE_INFINIBAND_H} = yes -a ${HAVE_LIBIBVERBS} = yes; then
+        HAVE_INFINIBAND=yes
+	ibverbs_LIBS="${ibverbs_LIBS} -libverbs"
+      else
+        HAVE_INFINIBAND=no
+        ibverbs_CPPFLAGS=
+    	ibverbs_LIBS=
+        if test "x${with_ibverbs}" != "xcheck"; then
+          AC_MSG_FAILURE([--with-ibverbs was given, but infiniband/verbs.h is not found])
+        fi
+      fi
+      CPPFLAGS="${save_CPPFLAGS}"
+      LIBS="${save_LIBS}"
+    fi    
+    AC_SUBST(HAVE_INFINIBAND)
+    AC_SUBST(ibverbs_CPPFLAGS)
+    AC_SUBST(ibverbs_LIBS)
+  ])
+
+dnl -- check for Intel PSM
+dnl --
+AC_DEFUN([AC_PADICO_CHECK_PSM],
+  [
+    HAVE_PSM=no
+    AC_ARG_WITH([psm],
+	        [AS_HELP_STRING([--with-psm],
+		 [use Intel Performance Scaled Messasging (PSM) @<:@default=check@:>@] )],
+	         [], [ with_psm=check ])
+    if test "x${with_psm}" != "xno"; then
+      save_CPPFLAGS="${CPPFLAGS}"
+      save_LIBS="${LIBS}"
+      if test "x${with_psm}" != "xyes" -a "x${with_psm}" != "xcheck"; then
+        PSM_DIR="${with_psm}"
+      fi
+      if test "x${PSM_DIR}" != "x"; then
+  	psm_CPPFLAGS="-I${PSM_DIR}/include"
+	psm_LIBS="-Wl,-rpath,${PSM_DIR}/lib -L${PSM_DIR}/lib"
+      fi
+      CPPFLAGS="${CPPFLAGS} ${psm_CPPFLAGS}"
+      LIBS="${LIBS} ${psm_LIBS}"
+      AC_CHECK_HEADER(psm.h, [ HAVE_PSM_H=yes ], [ HAVE_PSM_H=no ])
+      AC_SEARCH_LIBS(psm_init, [ psm psm_infinipath ], [ HAVE_LIBPSM=yes ], [ HAVE_LIBPSM=no ])
+      if test ${HAVE_PSM_H} = yes -a ${HAVE_LIBPSM} = yes; then
+        HAVE_PSM=yes
+	psm_LIBS="${psm_LIBS} ${ac_cv_search_psm_init}"
+      else
+	HAVE_PSM=no
+	psm_CPPFLAGS=
+	psm_LIBS=
+        if test "x${with_psm}" != "xcheck"; then
+      	  AC_MSG_FAILURE([--with-psm was given but psm.h is not found.])
+        fi
+      fi
+      CPPFLAGS="${save_CPPFLAGS}"
+      LIBS="${save_LIBS}"
+    fi
+    AC_SUBST(HAVE_PSM)
+    AC_SUBST([psm_CPPFLAGS])
+    AC_SUBST([psm_LIBS])
+  ])
+
+dnl -- check for Intel PSM2
+dnl --
+AC_DEFUN([AC_PADICO_CHECK_PSM2],
+  [
+    HAVE_PSM2=no
+    AC_ARG_WITH([psm2],
+		[AS_HELP_STRING([--with-psm2],
+	    	 [use Intel Performance Scaled Messasging 2 (PSM2) @<:@default=check@:>@] )],
+		 [], [ with_psm2=check ])
+    if test "x${with_psm2}" != "xno"; then
+      save_CPPFLAGS="${CPPFLAGS}"
+      save_LIBS="${LIBS}"
+      if test "x${with_psm2}" != "xyes" -a "x${with_psm2}" != "xcheck"; then
+        PSM2_DIR="${with_psm2}"
+      fi
+      if test "x${PSM2_DIR}" != "x"; then
+      	psm2_CPPFLAGS="-I${PSM2_DIR}/include"
+        psm2_LIBS="-Wl,-rpath,${PSM2_DIR}/lib -L${PSM2_DIR}/lib"
+      fi
+      CPPFLAGS="${CPPFLAGS} ${psm2_CPPFLAGS}"
+      LIBS="${LIBS} ${psm2_LIBS}"
+      AC_CHECK_HEADER(psm2.h, [ HAVE_PSM2_H=yes ], [ HAVE_PSM2_H=no ])
+      AC_CHECK_LIB(psm2, psm2_init, [ HAVE_LIBPSM2=yes ], [ HAVE_LIBPSM2=no ])
+      if test ${HAVE_PSM2_H} = yes -a ${HAVE_LIBPSM2} = yes; then
+        HAVE_PSM2=yes
+	psm2_LIBS="${psm2_LIBS} -lpsm2"
+      else
+	HAVE_PSM2=no
+	psm2_CPPFLAGS=
+	psm2_LIBS=
+        if test "x${with_psm2}" != "xcheck"; then
+      	  AC_MSG_FAILURE([--with-psm2 was given but psm2.h is not found.])
+        fi
+      fi
+      CPPFLAGS="${save_CPPFLAGS}"
+      LIBS="${save_LIBS}"
+    fi
+    AC_SUBST(HAVE_PSM2)
+    AC_SUBST([psm2_CPPFLAGS])
+    AC_SUBST([psm2_LIBS])
+  ])
+
+dnl -- check for OFI (libfabric)
+dnl --
+AC_DEFUN([AC_PADICO_CHECK_OFI],
+  [
+    HAVE_OFI=no
+    AC_ARG_WITH([ofi],
+		[AS_HELP_STRING([--with-ofi],
+	    	 [use OpenFabric Interface (libfabric) @<:@default=check@:>@] )],
+		 [], [ with_ofi=check ])
+    if test "x${with_ofi}" != "xno"; then
+      PKG_CHECK_MODULES([ofi], [libfabric], [HAVE_OFI=yes], [HAVE_OFI=no])
+    fi
+    if test "x${with_ofi}" = "xyes" -a "${HAVE_OFI}" = "no"; then
+      AC_MSG_ERROR([ required library OFI (libfabric) not found.])
+    fi
+    if test "${HAVE_OFI}" = "yes"; then
+      AC_CHECK_HEADER(rdma/fabric.h, [ HAVE_FABRIC_H=yes ], [ HAVE_FABRIC_H=no ])
+      requires_ofi=libfabric
+    fi
+    AC_SUBST([HAVE_OFI])
+    AC_SUBST([ofi_CFLAGS])
+    AC_SUBST([ofi_LIBS])
+  ])
+
+dnl --- Types
+dnl --------------------------------------------------------
+
+# ## socklen_t
+AC_DEFUN(AC_TYPE_SOCKLEN_T,
+[AC_CACHE_CHECK([for socklen_t], ac_cv_type_socklen_t,
+[
+  AC_TRY_COMPILE(
+  [#include <sys/socket.h>],
+  [socklen_t len = 42; return len;],
+  ac_cv_type_socklen_t=yes,
+  ac_cv_type_socklen_t=no)
+])
+  if test $ac_cv_type_socklen_t = yes; then
+    AC_DEFINE(HAVE_SOCKLEN_T, [], [whether socklen_t is defined in sys/socket.h])
+  fi
+])
+
+# ## off64_t
+AC_DEFUN([AC_TYPE_OFF64_T],
+	 [ AC_CACHE_CHECK([for off64_t], ac_cv_type_off64_t,
+[
+  AC_TRY_COMPILE(
+  [#define _LARGEFILE64_SOURCE
+   #include <sys/types.h>],
+  [off64_t off = 42;],
+  ac_cv_type_off64_t=yes,
+  ac_cv_type_off64_t=no)
+])
+  if test $ac_cv_type_off64_t = yes; then
+    AC_DEFINE(HAVE_OFF64_T, [], [whether off64_t is defined in sys/types.h using _LARGEFILE64_SOURCE])
+    largefilecflags=-D_LARGEFILE64_SOURCE
+  fi
+  AC_SUBST(largefilecflags)
+])
+
+# ## check base required types
+AC_DEFUN([AC_PADICO_TYPES],
+	 [ AC_REQUIRE([AC_PADICO_COMPILER])
+	   AC_C_BIGENDIAN
+	   AC_CHECK_SIZEOF(int)
+	   AC_CHECK_SIZEOF(long)
+	   AC_CHECK_SIZEOF(long long)
+	   AC_CHECK_SIZEOF(ssize_t)
+	   AC_TYPE_SOCKLEN_T
+	   AC_TYPE_OFF_T
+	   AC_TYPE_OFF64_T
+	   AC_TYPE_SIZE_T
+	   AC_TYPE_PID_T
+	 ])
+
+dnl --- libs
+dnl --------------------------------------------------------
+
+AC_DEFUN([AC_PADICO_LIBS],
+	 [ AC_REQUIRE([AC_PADICO_LINKER])
+	   AC_SEARCH_LIBS(dlopen,  [dl dlcompat c c++], [], [ AC_MSG_ERROR([Cannot find dlopen() ]) ])
+	   AC_SEARCH_LIBS(dlclose, [dl dlcompat c c++], [], [ AC_MSG_ERROR([Cannot find dlclose() ]) ])
+	   AC_SEARCH_LIBS(dlsym,   [dl dlcompat c c++], [], [ AC_MSG_ERROR([Cannot find dlsym() ]) ])
+
+	   AC_SEARCH_LIBS(sched_yield,   [c rt], [], [ AC_MSG_ERROR([Cannot find sched_yield() ]) ])
+
+	   AC_SEARCH_LIBS(socket, nsl socket, [], [ AC_MSG_ERROR([Cannot find socket() ]) ])
+	   AC_SEARCH_LIBS(listen, nsl socket, [], [ AC_MSG_ERROR([Cannot find listen() ]) ])
+	   AC_SEARCH_LIBS(accept, nsl socket, [], [ AC_MSG_ERROR([Cannot find accept() ]) ])
+	   AC_SEARCH_LIBS(gethostbyname, nsl socket, [], [ AC_MSG_ERROR([Cannot find gethostbyname() ]) ])
+   
+	 ])
+
+
+dnl --- Arguments
+dnl --------------------------------------------------------
+
+AC_DEFUN([AC_PADICO_OPTIMIZE],
+	 [ ENABLE_OPTIMIZE_FLAG="yes";
+	   AC_ARG_ENABLE(optimize,
+	        	 [  --enable-optimize       Build with optimization compiler flags @<:@default: yes@:>@ ],
+	                 [ if test $enableval = no; then
+	                     ENABLE_OPTIMIZE_FLAG="no";
+	                   fi
+	                 ],
+	                 [])
+	   AC_SUBST(ENABLE_OPTIMIZE_FLAG)
+	 ])
+
+AC_DEFUN([AC_PADICO_DEBUG],
+	 [ ENABLE_DEBUG_FLAG="no";
+	   AC_ARG_ENABLE(debug,
+	   	         [  --enable-debug          Build with debug flags @<:@default: no@:>@],
+	                 [ if test $enableval = yes; then 
+	                     ENABLE_DEBUG_FLAG="yes";
+	                   fi
+	                 ],
+	                 [])
+	   AC_SUBST(ENABLE_DEBUG_FLAG)
+ 	 ])
+
+AC_DEFUN([AC_PADICO_TRACE],
+	 [ AC_ARG_ENABLE(trace,
+	                 [ --enable-trace ], 
+	                 [ if test "x$enableval" = "xyes"; then
+	                     AC_DEFINE(PADICO_TRACE_ON, 1, [enable Padico trace system])
+	                   fi],
+	                 [])
+	 ])
+
Index: PadicoTM/building-tools/html_footer.html
===================================================================
--- PadicoTM/building-tools/html_footer.html	(nonexistent)
+++ PadicoTM/building-tools/html_footer.html	(copie de travail)
@@ -0,0 +1,2 @@
+</body>
+</html>
Index: PadicoTM/building-tools/html_header.html
===================================================================
--- PadicoTM/building-tools/html_header.html	(nonexistent)
+++ PadicoTM/building-tools/html_header.html	(copie de travail)
@@ -0,0 +1,24 @@
+
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN""http://www.w3.org/TR/REC-html40/loose.dtd">
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
+    <LINK HREF="../../common/style.css" REL="stylesheet" TYPE="text/css">
+    <LINK REL="icon" type="image/png" HREF="../../common/favicon.ico">
+    <LINK REL="shortcut icon" type="image/x-icon" HREF="../../common/favicon.ico">
+    <title>$projectname &mdash; $title</title>
+  </head>
+  
+<body>
+
+    <!-- Generated by Doxygen $doxygenversion -->
+
+    <div class="header">
+      <img class="logo" src="../../common/inr_logo_cherch_UK_coul.png" alt="">
+      <div class="header_text">
+	<h1>$projectname</h1>
+	<h2>Documentation</h2>
+      </div>
+    </div>
+    
+      
\ No newline at end of file
Index: PadicoTM/building-tools/misc.m4
===================================================================
--- PadicoTM/building-tools/misc.m4	(nonexistent)
+++ PadicoTM/building-tools/misc.m4	(copie de travail)
@@ -0,0 +1,101 @@
+dnl -*- mode: Autoconf;-*-
+
+# ## detect padico building-tools
+AC_DEFUN([AC_PADICO_BUILDING_TOOLS],
+	 [
+	   AC_MSG_CHECKING([for common building-tools])
+	   if test -r ${srcdir}/building-tools/common_rules.mk.in; then
+	     building_tools=${srcdir}/building-tools/
+	   elif test -r ${srcdir}/../building-tools/common_rules.mk.in; then
+	     building_tools=${srcdir}/../building-tools/
+ 	   else
+	     AC_MSG_ERROR([missing building-tools directory in srcdir: ${srcdir}])
+	   fi
+	   AC_MSG_RESULT([${building_tools}])
+	   AC_SUBST([building_tools])
+	 ])
+
+# ## set package name
+AC_DEFUN([AC_PADICO_PACKAGE],
+	 [
+	   AC_MSG_CHECKING([for package name])
+	   package_name="$1"
+	   package_short_name="$2"
+	   if test "x${package_short_name}" = "x"; then
+	     package_short_name="${package_name}"
+	   fi
+	   PACKAGE_NAME="${package_name}"
+	   if test "x${PACKAGE_NAME}" = "x"; then
+	     AC_MSG_ERROR([ cannot find package name. ])
+	   else
+	     AC_MSG_RESULT([${package_name} (${package_short_name})])
+	   fi
+	   AC_SUBST([package_name])
+	   AC_SUBST([package_short_name])
+	 ])
+
+# ## set package root
+AC_DEFUN([AC_PADICO_PACKAGE_ROOT],
+	 [
+	   AC_REQUIRE([AC_PADICO_PACKAGE])
+	   AC_MSG_CHECKING([for package root])
+           if test x$prefix = xNONE; then
+             AC_MSG_ERROR([No prefix given. Please supply a '--prefix' parameter.])
+           fi
+	   padico_root=${prefix}
+           AC_MSG_RESULT([${padico_root})])
+	   AC_SUBST([padico_root])
+	 ])
+
+# ## check build directory
+AC_DEFUN([AC_PADICO_BUILDDIR],
+	 [
+	   AC_REQUIRE([AC_PADICO_PACKAGE])
+	   AC_MSG_CHECKING([for package build directory])
+	   padico_builddir=`pwd`
+	   if test -r ./configure; then
+	     AC_MSG_ERROR([ *** ${PACKAGE_NAME} must not be configured in its source tree. Create a separate 'build' dir instead. ])
+	   fi
+	   AC_SUBST(padico_builddir)
+	   AC_MSG_RESULT(${padico_builddir})
+	 ])
+# ## check source directory
+AC_DEFUN([AC_PADICO_SRCDIR],
+	 [
+	   AC_REQUIRE([AC_PADICO_PACKAGE])
+	   AC_MSG_CHECKING([for package ${PACKAGE_NAME} source directory])
+	   case ${srcdir} in
+	     /*) padico_srcdir=${srcdir} ;;
+	     *)  padico_srcdir="`pwd`/${srcdir}" ;;
+	   esac
+	   AC_SUBST(padico_srcdir)
+	   AC_MSG_RESULT(${padico_srcdir})
+	 ])
+
+# ## Version and version strings
+AC_DEFUN([AC_PADICO_VERSION],
+	 [
+	   AC_REQUIRE([AC_PADICO_BUILDING_TOOLS])
+	   AC_REQUIRE([AC_PADICO_PACKAGE])
+           AC_REQUIRE([AC_PADICO_SRCDIR])
+           AC_MSG_CHECKING([${PACKAGE_NAME} version])
+	   if test -r ${building_tools}/VERSION; then
+	     PACKAGE_VERSION="`cat ${building_tools}/VERSION`"
+ 	   else
+	     AC_MSG_ERROR([ cannot find VERSION. ])
+           fi
+	   PACKAGE_STRING="${PACKAGE_NAME}-${PACKAGE_VERSION}"
+ 	   PACKAGE_TARNAME="${PACKAGE_STRING}.tar.gz"
+           PACKAGE_BUGREPORT="<Alexandre.Denis@inria.fr>"
+
+	   AC_MSG_RESULT([ $PACKAGE_STRING ])
+
+	   PADICO_VERSION=${PACKAGE_VERSION}
+	   AC_SUBST(PADICO_VERSION)
+
+           padico_build_string="${PACKAGE_STRING} (`date`)"
+	   AC_DEFINE_UNQUOTED(PADICO_BUILD_STRING, "${padico_build_string}", [date of the build])
+
+	   AC_DEFINE_UNQUOTED(PADICO_VERSION_STRING, "${PACKAGE_STRING}",
+			      [Padico version as string])
+	 ])
Index: PadicoTM/building-tools/out.m4
===================================================================
--- PadicoTM/building-tools/out.m4	(nonexistent)
+++ PadicoTM/building-tools/out.m4	(copie de travail)
@@ -0,0 +1,51 @@
+dnl -*- mode: Autoconf;-*-
+
+# ## add common makefiles to out list
+AC_DEFUN([AC_PADICO_OUT_COMMON_MK],
+	 [
+	   AC_REQUIRE([AC_PADICO_BASE_TOOLS])
+	   AC_REQUIRE([AC_PADICO_CDEP])
+	   AC_REQUIRE([AC_PADICO_LINKER])
+	   AC_REQUIRE([AC_PADICO_OPTIMIZE])
+	   AC_REQUIRE([AC_PADICO_DEBUG])
+	   AC_REQUIRE([AC_PADICO_BUILDING_TOOLS])
+	   out_mk="common_vars.mk:${building_tools}/common_vars.mk.in ${out_mk}"
+	   out_mk="common_rules.mk:${building_tools}/common_rules.mk.in ${out_mk}"
+	  ])
+
+# ## add root Makefile to out list
+AC_DEFUN([AC_PADICO_OUT_ROOT],
+	 [
+	   out_mk="${out_mk} Makefile"
+	   if test -r ${srcdir}/doc/Doxyfile.in; then
+  	     out_mk="${out_mk} doc/Doxyfile"
+	   fi
+	   if test -r ${srcdir}/${package_short_name}.pc.in; then
+	     out_mk="${out_mk} ${package_short_name}.pc"
+	   fi
+	 ])
+
+# ## add a shell script to out list
+AC_DEFUN([AC_PADICO_OUT_SH],
+	 [
+           AC_CONFIG_FILES( [ $1 ] , [ chmod +x $1 ])
+	 ])
+
+# ## issue output commands from out_mk lists
+AC_DEFUN([AC_PADICO_OUTPUT],
+	 [
+	   AC_CONFIG_FILES([ ${out_mk} ])
+	   for f in ${out_mk}; do
+	     case ${f} in
+	       *:*)
+	         src=`echo ${f} | cut -f 1 -d ':'`
+	       ;;
+	       *)
+	         src=${f}
+	       ;;
+	     esac
+	     PADICO_OUT_FILES="${PADICO_OUT_FILES} ${src}"
+	   done
+	   AC_SUBST([PADICO_OUT_FILES])
+	   AC_OUTPUT
+	 ])
Index: PadicoTM/building-tools/pkg.m4
===================================================================
--- PadicoTM/building-tools/pkg.m4	(nonexistent)
+++ PadicoTM/building-tools/pkg.m4	(copie de travail)
@@ -0,0 +1,157 @@
+# pkg.m4 - Macros to locate and utilise pkg-config.            -*- Autoconf -*-
+# serial 1 (pkg-config-0.24)
+# 
+# Copyright © 2004 Scott James Remnant <scott@netsplit.com>.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# As a special exception to the GNU General Public License, if you
+# distribute this file as part of a program that contains a
+# configuration script generated by Autoconf, you may include it under
+# the same distribution terms that you use for the rest of that program.
+
+# PKG_PROG_PKG_CONFIG([MIN-VERSION])
+# ----------------------------------
+AC_DEFUN([PKG_PROG_PKG_CONFIG],
+[m4_pattern_forbid([^_?PKG_[A-Z_]+$])
+m4_pattern_allow([^PKG_CONFIG(_PATH)?$])
+AC_ARG_VAR([PKG_CONFIG], [path to pkg-config utility])
+AC_ARG_VAR([PKG_CONFIG_PATH], [directories to add to pkg-config's search path])
+AC_ARG_VAR([PKG_CONFIG_LIBDIR], [path overriding pkg-config's built-in search path])
+
+if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
+	AC_PATH_TOOL([PKG_CONFIG], [pkg-config])
+fi
+if test -n "$PKG_CONFIG"; then
+	_pkg_min_version=m4_default([$1], [0.9.0])
+	AC_MSG_CHECKING([pkg-config is at least version $_pkg_min_version])
+	if $PKG_CONFIG --atleast-pkgconfig-version $_pkg_min_version; then
+		AC_MSG_RESULT([yes])
+	else
+		AC_MSG_RESULT([no])
+		PKG_CONFIG=""
+	fi
+fi[]dnl
+])# PKG_PROG_PKG_CONFIG
+
+# PKG_CHECK_EXISTS(MODULES, [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])
+#
+# Check to see whether a particular set of modules exists.  Similar
+# to PKG_CHECK_MODULES(), but does not set variables or print errors.
+#
+# Please remember that m4 expands AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+# only at the first occurence in configure.ac, so if the first place
+# it's called might be skipped (such as if it is within an "if", you
+# have to call PKG_CHECK_EXISTS manually
+# --------------------------------------------------------------
+AC_DEFUN([PKG_CHECK_EXISTS],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
+if test -n "$PKG_CONFIG" && \
+    AC_RUN_LOG([$PKG_CONFIG --exists --print-errors "$1"]); then
+  m4_default([$2], [:])
+m4_ifvaln([$3], [else
+  $3])dnl
+fi])
+
+# _PKG_CONFIG([VARIABLE], [COMMAND], [MODULES])
+# ---------------------------------------------
+m4_define([_PKG_CONFIG],
+[if test -n "$$1"; then
+    pkg_cv_[]$1="$$1"
+ elif test -n "$PKG_CONFIG"; then
+    PKG_CHECK_EXISTS([$3],
+                     [pkg_cv_[]$1=`$PKG_CONFIG --[]$2 "$3" 2>/dev/null`],
+		     [pkg_failed=yes])
+ else
+    pkg_failed=untried
+fi[]dnl
+])# _PKG_CONFIG
+
+# _PKG_SHORT_ERRORS_SUPPORTED
+# -----------------------------
+AC_DEFUN([_PKG_SHORT_ERRORS_SUPPORTED],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
+        _pkg_short_errors_supported=yes
+else
+        _pkg_short_errors_supported=no
+fi[]dnl
+])# _PKG_SHORT_ERRORS_SUPPORTED
+
+
+# PKG_CHECK_MODULES(VARIABLE-PREFIX, MODULES, [ACTION-IF-FOUND],
+# [ACTION-IF-NOT-FOUND])
+#
+#
+# Note that if there is a possibility the first call to
+# PKG_CHECK_MODULES might not happen, you should be sure to include an
+# explicit call to PKG_PROG_PKG_CONFIG in your configure.ac
+#
+#
+# --------------------------------------------------------------
+AC_DEFUN([PKG_CHECK_MODULES],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
+AC_ARG_VAR([$1][_CFLAGS], [C compiler flags for $1, overriding pkg-config])dnl
+AC_ARG_VAR([$1][_LIBS], [linker flags for $1, overriding pkg-config])dnl
+
+pkg_failed=no
+AC_MSG_CHECKING([for $1])
+
+_PKG_CONFIG([$1][_CFLAGS], [cflags], [$2])
+_PKG_CONFIG([$1][_LIBS], [libs], [$2])
+
+m4_define([_PKG_TEXT], [Alternatively, you may set the environment variables $1[]_CFLAGS
+and $1[]_LIBS to avoid the need to call pkg-config.
+See the pkg-config man page for more details.])
+
+if test $pkg_failed = yes; then
+   	AC_MSG_RESULT([no])
+        _PKG_SHORT_ERRORS_SUPPORTED
+        if test $_pkg_short_errors_supported = yes; then
+	        $1[]_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "$2" 2>&1`
+        else 
+	        $1[]_PKG_ERRORS=`$PKG_CONFIG --print-errors "$2" 2>&1`
+        fi
+	# Put the nasty error message in config.log where it belongs
+	echo "$$1[]_PKG_ERRORS" >&AS_MESSAGE_LOG_FD
+
+	m4_default([$4], [AC_MSG_ERROR(
+[Package requirements ($2) were not met:
+
+$$1_PKG_ERRORS
+
+Consider adjusting the PKG_CONFIG_PATH environment variable if you
+installed software in a non-standard prefix.
+
+_PKG_TEXT])dnl
+        ])
+elif test $pkg_failed = untried; then
+     	AC_MSG_RESULT([no])
+	m4_default([$4], [AC_MSG_FAILURE(
+[The pkg-config script could not be found or is too old.  Make sure it
+is in your PATH or set the PKG_CONFIG environment variable to the full
+path to pkg-config.
+
+_PKG_TEXT
+
+To get pkg-config, see <http://pkg-config.freedesktop.org/>.])dnl
+        ])
+else
+	$1[]_CFLAGS=$pkg_cv_[]$1[]_CFLAGS
+	$1[]_LIBS=$pkg_cv_[]$1[]_LIBS
+        AC_MSG_RESULT([yes])
+	$3
+fi[]dnl
+])# PKG_CHECK_MODULES
Index: PadicoTM/building-tools/tools.m4
===================================================================
--- PadicoTM/building-tools/tools.m4	(nonexistent)
+++ PadicoTM/building-tools/tools.m4	(copie de travail)
@@ -0,0 +1,166 @@
+dnl -*- mode: Autoconf;-*-
+
+# ## basic tools
+AC_DEFUN([AC_PADICO_BASE_TOOLS],
+	 [
+	   dnl --- true
+	   AC_PATH_PROG(BIN_TRUE, true)
+	   AC_SUBST(BIN_TRUE)
+
+	   dnl --- bash
+	   AC_PATH_PROG([BASH], [bash])
+	   AC_SUBST([BASH])
+	])
+
+AC_DEFUN([AC_PADICO_CHECK_GNUPLOT],
+	 [
+	   AC_PATH_PROG(GNUPLOT, gnuplot)
+	   AC_SUBST(GNUPLOT)
+	 ])
+	
+AC_DEFUN([AC_PADICO_CHECK_XTERM],
+	 [
+	   dnl --- xterm
+	   if test "x" = "x$XTERM" ; then
+	     AC_PATH_PROGS(xterm_path, [ xterm x-terminal-emulator rxvt Eterm konsole ])
+	     XTERM="$xterm_path"
+	   else
+	     XTERM="$XTERM"
+	   fi
+	   AC_ARG_VAR(XTERM, [ The X Window terminal emulator ])
+	   if test "x$XTERM" = "x"; then
+	     AC_MSG_WARN([xterm not found.])
+           else
+             AC_DEFINE_UNQUOTED(XTERM, "${XTERM}", [ X Window terminal emulator ])
+ 	   fi
+	])
+
+AC_DEFUN([AC_PADICO_CHECK_DIALOG],
+	 [
+	   dnl --- dialog
+	   AC_PATH_PROGS(DIALOG, [ ${DIALOG} whiptail dialog cdialog gdialog ])
+	   AC_ARG_VAR(DIALOG, [ The program used to display dialogs (e.g. whiptail) ])
+	 ])
+
+AC_DEFUN([AC_PADICO_CHECK_GDB],
+	 [
+	   dnl --- gdb
+	   AC_PATH_PROGS(GDB, [ ${GDB} gdb ])
+	   AC_ARG_VAR(GDB, [ The gdb debugger ])
+	   AC_DEFINE_UNQUOTED(GDB, "${GDB}", [ The gdb debugger ])
+	   if test "x$GDB" = "x"; then
+	     AC_MSG_WARN([gdb not found.])
+ 	   fi
+	 ])
+
+# ## check for GNU make
+AC_DEFUN([AC_PADICO_GNU_MAKE],
+	 [ AC_CACHE_CHECK( for GNU make,_cv_gnu_make_command,
+                _cv_gnu_make_command='' ;
+dnl Search all the common names for GNU make
+                for a in "$MAKE" make gmake gnumake gnu-make ; do
+                        if test -z "$a" ; then continue ; fi ;
+                        if  ( sh -c "$a --version" 2> /dev/null | grep GNU  2>&1 > /dev/null ) ;  then
+                                _cv_gnu_make_command=$a ;
+                                break;
+                        fi
+                done ;
+        ) ;
+dnl If there was a GNU version, then set @ifGNUmake@ to the empty string, '#' otherwise
+        if test  "x$_cv_gnu_make_command" != "x"  ; then
+                MAKE=$_cv_gnu_make_command
+                ifGNUmake='' 
+        else
+                ifGNUmake='#'
+                AC_MSG_ERROR([Not found])
+        fi
+        AC_ARG_VAR(MAKE, [ path for GNU make ])
+        AC_SUBST(ifGNUmake)
+	 ] )
+
+# ##  C/C++ dependency checker
+AC_DEFUN([AC_PADICO_CDEP],
+	 [ AC_REQUIRE([AC_PADICO_COMPILER])
+	   AC_MSG_CHECKING([how to check C dependency])
+	   CC_DEP="${CC} ${CDEPFLAGS}"
+	   AC_MSG_RESULT([${CC_DEP}])
+	   AC_SUBST(CC_DEP)
+	 ])
+
+AC_DEFUN([AC_PADICO_CXXDEP],
+	[ AC_REQUIRE([AC_PADICO_CXX])
+	   AC_MSG_CHECKING([how to check C++ dependency])
+	   CXX_DEP="${CXX} ${CXXDEPFLAGS}"
+	   AC_MSG_RESULT([${CXX_DEP}])
+	   AC_SUBST(CXX_DEP)
+	 ])	
+
+# ## static linker
+AC_DEFUN([AC_PADICO_LINKER],
+	 [ AC_REQUIRE([AC_PADICO_COMPILER])
+	   AC_MSG_CHECKING([for static linker])
+	   LD="${CC}"
+	   AC_MSG_RESULT([${LD} ${LDFLAGS}])
+	   AC_SUBST(LD)
+	   AC_SUBST(LDFLAGS)
+	 ])
+
+# ## dynamic linker
+AC_DEFUN([AC_PADICO_DYNLD],
+	 [ AC_REQUIRE([AC_PADICO_COMPILER])
+	   AC_MSG_CHECKING([for dynamic library linker])
+	   DYNLIB_LD="${CC} ${LDFLAGS} ${CPICFLAGS} ${DYNLIB_FLAGS}"
+	   AC_MSG_RESULT([${DYNLIB_LD}])
+	   AC_SUBST(DYNLIB_LD)
+
+	   AC_MSG_CHECKING([for dynamic library prefix])
+	   AC_MSG_RESULT([${DYNLIB_PREFIX}])
+	   AC_SUBST(DYNLIB_PREFIX)
+	   AC_DEFINE_UNQUOTED(DYNLIB_PREFIX, "${DYNLIB_PREFIX}",
+	     [prefix prepended to dynamic library names (usually "lib")])
+
+	   AC_MSG_CHECKING([for dynamic library extension])
+	   AC_MSG_RESULT([${DYNLIB_EXT}])
+	   AC_SUBST(DYNLIB_EXT)
+	   AC_DEFINE_UNQUOTED(DYNLIB_EXT, "${DYNLIB_EXT}",
+	     [suffix for dynamic library names (usually ".so" on Unix systems)])
+	   HAVE_DYNLIB=yes
+	   AC_SUBST([HAVE_DYNLIB])
+	 ])
+
+AC_DEFUN([AC_PADICO_EXPAT],
+	 [
+	   AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+	   AC_ARG_VAR(EXPAT_ROOT, "Specify Expat location in the system (default: use pkg-config)")
+	   if test "x$EXPAT_ROOT" != "x"; then
+ 	       BASE_CFLAGS="$BASE_CFLAGS -I$EXPAT_ROOT/include"
+ 	       CPPFLAGS="$CPPFLAGS -I$EXPAT_ROOT/include"
+	       LDFLAGS="$LDFLAGS -L$EXPAT_ROOT/lib"
+	   else
+ 	     PKG_CHECK_MODULES([expat], [ expat ])
+       	     BASE_CFLAGS="$BASE_CFLAGS $expat_CFLAGS"
+ 	     CPPFLAGS="$CPPFLAGS $expat_CFLAGS"
+	     LIBS="$LIBS $expat_LIBS"
+	     requires_expat=expat
+	     AC_SUBST([requires_expat])
+	   fi
+	   AC_CHECK_HEADERS(expat.h, 
+	   		    [ HAVE_EXPAT_H=yes ],
+	     		    [ HAVE_EXPAT_H=no  ])
+	   if test $HAVE_EXPAT_H = "no"; then
+	       AC_MSG_ERROR([ Expat XML parser not found ])
+	   fi
+	 ])
+
+
+dnl -- detect Puk as external tool
+AC_DEFUN([AC_PADICO_PUK_ROOT],
+	 [
+	     AC_MSG_CHECKING([for Puk root])
+	     puk_root="`pkg-config --variable=prefix Puk`"
+	     AC_SUBST([puk_root])
+	     AC_MSG_RESULT([${puk_root}])
+	     if test ! -f ${puk_root}/etc/common_rules.mk; then
+	         AC_MSG_ERROR([required ${puk_root}/etc/common_rules.mk not found])
+             fi
+	 ])
