--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -46,7 +46,7 @@ foreach(_file ${MAPHYS_EXAMPLES})
   get_filename_component(_name_exe ${_file} NAME_WE)
   add_executable(${_name_exe} ${_file})
   target_link_libraries(${_name_exe} maphys mph_toolkit packcg packgmres slatec ${MAPHYS_EXTRA_LIBRARIES})
-  install(TARGETS ${_name_exe} DESTINATION ${CMAKE_INSTALL_PREFIX}/examples)
+  install(TARGETS ${_name_exe} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/maphys/examples)
 endforeach()
 
 # install input files for examples
@@ -76,7 +76,7 @@ if(MAPHYS_ORDERING_PADDLE)
     )
 endif()
 
-install(FILES ${MAPHYS_EXAMPLES_INPUT_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX}/examples)
+install(FILES ${MAPHYS_EXAMPLES_INPUT_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/maphys/examples)
 
 # Add tests to CTest
 #TODO: make those test more robust (check the solution)
diff --git a/gendistsys/CMakeLists.txt b/gendistsys/CMakeLists.txt
index 125be7e..0659e95 100644
--- a/gendistsys/CMakeLists.txt
+++ b/gendistsys/CMakeLists.txt
@@ -11,7 +11,7 @@ foreach(_file ${MAPHYS_TESTGENDIST})
   get_filename_component(_name_exe ${_file} NAME_WE)
   add_executable(${_name_exe} ${_file})
   target_link_libraries(${_name_exe} maphys mph_toolkit ${MAPHYS_EXTRA_LIBRARIES})
-  install(TARGETS ${_name_exe} DESTINATION ${CMAKE_INSTALL_PREFIX}/gendistsys)
+  install(TARGETS ${_name_exe} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/maphys/gendistsys)
 endforeach()
 
 # install input files for gendistsys
@@ -21,4 +21,4 @@ set(MAPHYS_TESTGENDIST_INPUT_FILES
   gds.in
   )
 
-install(FILES ${MAPHYS_TESTGENDIST_INPUT_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX}/gendistsys)
+install(FILES ${MAPHYS_TESTGENDIST_INPUT_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/maphys/gendistsys)
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 8947caf..ee5d316 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -240,7 +240,7 @@ install(TARGETS maphys
 # Copy the fortran module files
 
 install ( CODE
-  "EXECUTE_PROCESS (COMMAND \"${CMAKE_COMMAND}\" -E copy_directory \"${CMAKE_Fortran_MODULE_DIRECTORY}/\${BUILD_TYPE}\" \"${CMAKE_INSTALL_PREFIX}/modules\")"
+  "EXECUTE_PROCESS (COMMAND \"${CMAKE_COMMAND}\" -E copy_directory \"${CMAKE_Fortran_MODULE_DIRECTORY}/\${BUILD_TYPE}\" \"${CMAKE_INSTALL_PREFIX}/include\")"
   )
 
 ###
